### ISSUE TITLE
Recent clang fails to compile lz4.h

### ISSUE THREAD

**Issue by Leandros** [lz4/lz4#409]
If C++14 is enabled, the `[[deprecated("message")]]` attribute will be applied to deprecated functions. If now the compiler is also defining `__GNUC__` as higher than 4, `LZ4LIB` will be defined to `__attribute__ ((__visibility__ ("default")))` which will raise the following errors:

```
37 : <source>:37:12: error: an attribute list cannot appear here
LZ4LIB_API LZ4_DEPRECATED("use LZ4_compress_default() instead") int LZ4_compress               (const char* source, char* dest, int sourceSize)
```

Can be fixed for the time being by defining `LZ4_DISABLE_DEPRECATE_WARNINGS`, but this should still be fixed.

**Comment by terrelln** [comment]
Hi @leandros, I believe this is already fixed in the `dev` branch by PR #393.

**Comment by Leandros** [comment]
I was suspecting it, but couldn't find it. Will close this!
